src/obstacle_avoidance/src/OlafNode.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Created By: Gareth Ellis
+ * Created On: July 16th, 2016
+ * Modified By: Marinah Zhao
+ * Modified On: Feb 12, 2018
+ * Description: A node that subscribes to a lidar topic publishing twist messages for the robot to move.
+ */
+
+#include <OlafNode.h>
+
+OlafClass::OlafClass(int argc, char **argv, std::string node_name) {
+
+    // Setup NodeHandles
+    ros::init(argc, argv, node_name);
+    ros::NodeHandle nh;
+    ros::NodeHandle private_nh("~");
+
+    // Setup Subscriber(s)
+    std::string topic_to_subscribe_to = "scan";
+    int queue_size = 1;
+    my_subscriber = nh.subscribe(topic_to_subscribe_to, queue_size, &OlafClass::lidarSubscriberCallBack, this);
+
+    // Setup Publisher(s)
+    std::string topic = private_nh.resolveName("/cmd_vel");
+    my_publisher = private_nh.advertise<geometry_msgs::Twist>(topic, queue_size);
+}
+
+void OlafClass::lidarSubscriberCallBack(sensor_msgs::LaserScan scan_msg) {
+
+    ROS_INFO("Received message from lidar");
+    ros::Time time = scan_msg.header.stamp;
+    float angle_min = scan_msg.angle_min;
+    float angle_max = scan_msg.angle_max;
+    float angle_increment = scan_msg.angle_increment;
+    float range_min = scan_msg.range_min;
+    float range_max = scan_msg.range_max;
+    std::vector<float> ranges = scan_msg.ranges;
+
+    int max_hole_size = 0;            // size of hole in number of indexes (number of angle increments)
+    int max_hole_start_index = 0;     // start index (angle) of the start of a hole
+    int cur_hole_size = 0;
+    int cur_hole_start_index = 0;
+
+    // Iterate through the vector
+    // Remove values out of range
```

Comment:
What does `range` mean here? Within the range of valid distances that the lidar can read? We should make sure we're explaining this clearly here.

Reply:
Done.